src/main/java/gui/LoadDataSet.java

```diff
@@ -0,0 +1,56 @@
+package gui;
+
+import detail.FileWrapper;
+import detail.LoadDatatSetTask;
+import detail.JStatGuiGlobalData;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.validation.Valid;
+import java.io.File;
+
+@Controller
+@RequestMapping("/load-dataset")
+public class LoadDataSet {
+
+    @GetMapping
+    public String loadDataSet(Model model){
```

Comment:
maybe some of the initialisations from main could happen here.

Reply:
No I disagree, the initializations we do are not only about datasets...so we better have all the initializations in one place rather than scattered....I am planning to have  a ```config``` file for providing some customization....

Reply:
I' d like to see the `config` file to understand what you mean. overall I agree with your point.